Update Redis & Database support info (#2502)
TykTechnologies/tyk-docs: tyk-docs/content/tyk-on-premises.md

### Tyk Dashboard

> Tyk Dashboard reuqires a persistent datastore for its operations. By default MongoDB is used. From Tyk v4.0, we also support PostgreSQL. See [Database Options]({{< ref "/content/tyk-dashboard/database-options.md" >}}) for list of versions and drop-in replacements we support.

Reviewer: ```suggestion
Tyk Dashboard reuqires a persistent datastore for its operations. By default MongoDB is used. From Tyk v4.0, we also support PostgreSQL. See [Database Options]({{< ref "tyk-dashboard/database-options.md" >}}) for a list of versions and drop-in replacements we support.
```

* Special notes for DocumentDB:

{{< note success >}}
**Note**  

If you are using DocumentDB, [capped collections]({{< ref "tyk-stack/tyk-manager/analytics/capping-analytics-data-storage" >}}) are not supported. See [here](https://docs.aws.amazon.com/documentdb/latest/developerguide/mongo-apis.html) for more details.
{{< /note >}}

> * Special notes for AtlasDB:

Reviewer: link

Visit the [Gateway page]({{< ref "tyk-oss-gateway" >}}) for more info.

> Tyk Gateway requires Redis for its operations. See [Redis]({{< ref "planning-for-production/redis#supported-versions">}}) for list of versions we support.

Reviewer: ```suggestion
Tyk Gateway requires Redis for its operations. See [supported Redis versions]({{< ref "planning-for-production/redis#supported-versions">}}) for the list of releases.
```

> Tyk Gateway requires Redis for its operations. See [Redis]({{< ref "/planning-for-production/redis/#supported-versions">}}) for list of versions we support.

Reviewer: ```suggestion
Tyk Gateway requires Redis for its operations. See [Redis]({{< ref "planning-for-production/redis#supported-versions">}}) for list of versions we support.
```
